
# HG changeset patch
# User Antoine Pitrou <solipsis@pitrou.net>
# Date 1415109181 -3600
# Node ID be374b8c40c896507ec36664789b12ff4eb3518b
# Parent  205810ca5069879b35c8efa14ceb8d6802a2ba01# Parent  c4b5a5d44254f619e469e6e873d74554ba77e703
Issue #22773: fix failing test with old readline versions due to issue #19884.

diff --git a/Lib/test/test_readline.py b/Lib/test/test_readline.py
--- a/Lib/test/test_readline.py
+++ b/Lib/test/test_readline.py
@@ -44,6 +44,10 @@ class TestHistoryManipulation (unittest.
 
 
 class TestReadline(unittest.TestCase):
+
+    @unittest.skipIf(readline._READLINE_VERSION < 0x0600
+                     and "libedit" not in readline.__doc__,
+                     "not supported in this library version")
     def test_init(self):
         # Issue #19884: Ensure that the ANSI sequence "\033[1034h" is not
         # written into stdout when the readline module is imported and stdout
diff --git a/Modules/readline.c b/Modules/readline.c
--- a/Modules/readline.c
+++ b/Modules/readline.c
@@ -1285,5 +1285,9 @@ PyInit_readline(void)
     mod_state = (readlinestate *) PyModule_GetState(m);
     PyOS_ReadlineFunctionPointer = call_readline;
     setup_readline(mod_state);
+
+    PyModule_AddIntConstant(m, "_READLINE_VERSION", RL_READLINE_VERSION);
+    PyModule_AddIntConstant(m, "_READLINE_RUNTIME_VERSION", rl_readline_version);
+
     return m;
 }

